src/model/client.ts

```diff
@@ -32,31 +45,47 @@ export class CommandoClient extends Client {
    * @param callback The function to call on the message
    */
   public onCommand(callback: (msg: CommandoMessage, cmd: Command, prefix: string) => void): void {
-    this.on('command', callback)
+    this.on(Event.COMMAND_RUN, callback)
   }
 
   /**
    * Calls a callback when a message starting with a prefix but not matching a command is sent.
    * @param callback The function to call on the invalid command
    */
   public onInvalidCommand(callback: (msg: CommandoMessage) => void): void {
-    this.on('invalidCommand', callback)
+    this.on(Event.INVALID_COMMAND, callback)
   }
 
   /**
    * Register a command object
    * @param command - The command object
    */
   public registerCommand(command: Command): void {
-    if (this.commands.has(command.options.name)) {
-      throw new Error(
+    const duplicate = this.findDuplicate(command)
+
+    if (duplicate) {
+      throw new TypeError(
         `Command '${
-          command.options.name
-        }' is already registered. Do you have two commands with the same name?`
+          duplicate.options.name
```

Comment:
Since this method will usually be called from the `registerCommandsIn` method, the user won't know which command was being registered.

Can this be changed to `Unable to register command '${command.options.name}'. I've already registered a command '${duplicate.options.name}' which either has the same name or shares an alias.`